### PR TITLE
Finished check_arg_count + added utils + added const to putstr

### PR DESCRIPTION
**Pull request by ArthurBeznik**
Maybe I should wait for this PR, and rewrite the Makefile before. What do you think ?

### CODE REVIEW

File: srcs/utils/exit_error.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * 	exit_error(3, "main", "malloc failure");
+ * 		should output to standard error:
+ * 			"main: malloc failure" 
+ * 		or 
+ * 	exit_error(errno, "main", NULL);
+ * 		should output to standard error:
+ * 			main: <errno message>
+ * 	both functions exit with given exit code after displaying error message
+*/
+static void	st_write_msg(const char *fn, const char *msg)
```
Comment (Zangetsu89) [inline srcs/utils/exit_error.c:26]: Maybe name the function logger()

File: srcs/parser/parser.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/*   Updated: 2023/02/12 17:39:38 by abeznik       ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../includes/parser.h"
+
+int	parser(int argc, char *argv[])
+{
+	if (argc != 2)
+		return (EXIT_FAILURE);
```
Comment (Zangetsu89) [inline srcs/parser/parser.c:18]: I would return true or false

File: srcs/main.c [hunk 3]
```diff
@@ -6,11 +6,18 @@
...
-/*   Updated: 2023/02/12 15:05:28 by arthurbezni   ########   odam.nl         */
+/*   Updated: 2023/02/12 17:38:06 by abeznik       ########   odam.nl         */
 /*                                                                            */
 /* ************************************************************************** */
 
+#include "../includes/cub3D.h"
+
 int	main(int argc, char *argv[])
 {
+	if (parser(argc, argv))
```
Comment (Zangetsu89) [inline srcs/main.c:17]: Better true or false return from parser. right now the statement doesn't make sense, if parser is true then return error.

**Review by Zangetsu89 — CHANGES_REQUESTED** [review]

File: srcs/parser/parser.c (hunk 2 above)
Comment (ArthurBeznik) [reply srcs/parser/parser.c:18]: i truly hate booleans

File: srcs/utils/exit_error.c (hunk 1 above)
Comment (ArthurBeznik) [reply srcs/utils/exit_error.c:26]: smart

**Comment by Zangetsu89** [comment]
Now ideally, Makefile shouldn't be related to this PR/task. if you make a change to Makefile then make it in a separate branch.

File: srcs/main.c (hunk 3 above)
Comment (ArthurBeznik) [reply srcs/main.c:17]: i hate bool but i'll change it to bool

Comment (Zangetsu89) [reply srcs/main.c:17]: I mean we can also customize our own codes if you really hate bools but in way that's logical

File: srcs/utils/exit_error.c [hunk 4]
```diff
@@ -43,9 +43,16 @@ static void	st_write_msg(const char *fn, const char *msg)
...
 {
 	if (function || message)
 		st_write_msg(function, message);
 	return (code);
 }
+
+int	exit_error(const char *message)
+{
+	if (message)
+		st_write_msg(NULL, message);
```
Comment (Zangetsu89) [inline srcs/utils/exit_error.c:56]: Logger() was supposed to replace st_write_msg right?

Comment (ArthurBeznik) [reply srcs/utils/exit_error.c:56]: So there's logger and exit error, which both use st_write_error to output either a message either a function name and/or a message

Comment (ArthurBeznik) [reply srcs/utils/exit_error.c:56]: But we might not need both as this was used for my minishell to get an appropriate error code and message

Comment (Zangetsu89) [reply srcs/utils/exit_error.c:56]: yeah this is fine, but writing the msg is should be logger, and the exit_error is for exit code, maybe exit_code_error()

**Review by Zangetsu89 — APPROVED** [review]

**Pull request merged**